ophyd/ophydobj.py

```diff
@@ -31,21 +45,28 @@ class OphydObject:
     def __init__(self, *, name=None, parent=None):
         super().__init__()
 
+        # base name and ref to parent, these go with properties
+        if name is None:
+            name = ''
         self._name = name
         self._parent = parent
 
-        # find magic class attributes to name 'event' types.
-        # for publicly exposed event types
-        self._pub_event_types = tuple(getattr(self, k) for k in dir(self) if
-                                      k.startswith('SUB_'))
-        # and for private event types
-        self._priv_event_types = tuple(getattr(self, k) for k in dir(self) if
-                                       k.startswith('_SUB_'))
-
-        self._subs = {k: [] for k in
-                      self._pub_event_types + self._priv_event_types}
-
-        self._sub_cache = defaultdict(lambda: None)
+        self.subscriptions = {getattr(self, k)
+                              for k in dir(type(self))
+                              if (k.startswith('SUB') or
+                                  k.startswith('_SUB'))}
+
+        # dictionary of wrapped callbacks
+        self._callbacks = {k: {} for k in self.subscriptions}
+        # this is to maintain api on clear_sub
+        self._unwrapperd_callbacks = {k: {} for k in self.subscriptions}
```

Comment:
I had to pick on private variable names, but can we do "unwrapped" here instead of "unwrapperd"?

Reply:
I no longer remember if that is a typo that editor helpful complete propogated or if it was intentional.